src/submission/models.py

```diff
@@ -1125,11 +1118,37 @@ def workflow_stages(self):
     @property
     def current_workflow_element(self):
         from core import models as core_models
-        logs = core_models.WorkflowLog.objects.filter(
-            article=self,
-        )
-        element = logs.reverse().first().element
-        return element
+        try:
+            workflow_element_name = workflow.STAGES_ELEMENTS.get(
+                self.stage,
+            )
+            return core_models.WorkflowElement.objects.get(
+                journal=self.journal,
+                element_name=workflow_element_name,
+            )
+        except (KeyError, core_models.WorkflowElement.DoesNotExist):
+            return None
+
+    @property
+    def current_workflow_element_url(self):
+        kwargs = {'article_id': self.pk}
+        # STAGE_UNASSIGNED and STAGE_PUBLISHED arent elements so are hardcoded.
+        if self.stage == STAGE_UNASSIGNED:
+            return reverse('review_unassigned_article', kwargs=kwargs)
+        elif self.stage == STAGE_PUBLISHED:
+            return reverse('manage_archive_article', kwargs=kwargs)
+        else:
+            element = self.current_workflow_element
+            if element:
+                return reverse(element.jump_url, kwargs=kwargs)
+            else:
+                logger.error(
+                    'Article #{} on Journal {} has no Stage.'.format(
```

Comment:
There are two different problems being obfuscated here:

- Article has no stage (can be ammended by a staff member)

- current_workflow_element returns None: Probably a bug that needs patching.
We should probably log that accordingly and tweak the nostage page as well